cmd/dump.go

```diff
@@ -79,13 +83,27 @@ func runDump(ctx *cli.Context) error {
 		os.Setenv("TMPDIR", tmpWorkDir)
 	}
 
-	reposDump := path.Join(tmpWorkDir, "gitea-repo.zip")
 	dbDump := path.Join(tmpWorkDir, "gitea-db.sql")
 
-	log.Printf("Dumping local repositories...%s", setting.RepoRootPath)
+	fileName := fmt.Sprintf("gitea-dump-%d.zip", time.Now().Unix())
+	log.Printf("Packing dump files...")
+	z, err := zip.Create(fileName)
+	if err != nil {
+		log.Fatalf("Failed to create %s: %v", fileName, err)
+	}
 	zip.Verbose = ctx.Bool("verbose")
-	if err := zip.PackTo(setting.RepoRootPath, reposDump, true); err != nil {
-		log.Fatalf("Failed to dump local repositories: %v", err)
+
+	if ctx.IsSet("skip-repository") {
+		log.Printf("Skip dumping local repositories")
+	} else {
+		log.Printf("Dumping local repositories...%s", setting.RepoRootPath)
+		reposDump := path.Join(tmpWorkDir, "gitea-repo.zip")
+		if err := zip.PackTo(setting.RepoRootPath, reposDump, true); err != nil {
+			log.Fatalf("Failed to dump local repositories: %v", err)
+		}
+		if err := z.AddFile("gitea-repo.zip", reposDump); err != nil {
```

Comment:
Why are you packing the repos zip into the dump zip?

The repo zip is likely to be large and this step will unnecessarily double the temporary size. Ensuring both files have the same timestamp might be better.

Reply:
@zeripath this line is for compatible with old implementation.

Reply:
Oh yeah ignore me.